frontend/app/routes/_public+/apply+/_index.tsx

```diff
@@ -65,6 +65,8 @@ export default function ApplyIndex() {
 
       captchaRef.current.resetCaptcha();
     }
+
+    sessionStorage.setItem('flow.state', 'active');
```

Comment:
I think the `sessionStorage.setItem` and navigate should happened after the action was successful.

`action` should return `json` with the generated `id`.
```
return json({ id }, sessionResponseInit);
```

then react when `useActionData` have an `id`.  Set the sessionStorage, then redirect client-side.

Reply:
I'll give it a shot.

Reply:
I've been thinking about this a little bit and I don't like the idea because it violates the [Post/Redirect/Get](https://en.wikipedia.org/wiki/Post/Redirect/Get) pattern that we use to avoid the back-button-post problem.

Also, it would completely break non-javascript clients. I acknowledge that we decided to require javascript, but I do think we should try to maintain some functionality for clients who happen to have javascript disabled.